app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
+      Minion.transaction do
+        Minion.find_or_create_by!(hostname: minion).assign_role(available_roles)
```

Comment:
But what if we don't have a minion with this hostname? In that case you are creating it right here which is not a responsibility of this method (because it might require a number of other actions which will be implemented [somewhere else](https://github.com/SUSE/pharos/pull/5/files#diff-784d5f9c206cafc566e70b874a69b0da)).